### DISCUSSION TITLE
Invalid API KEy

### DISCUSSION THREAD

**Discussion by akhilparakka** [smartcontractkit/full-blockchain-solidity-course-py#1600] [Q&A] [answered]
![Screenshot from 2022-07-25 12-32-14](https://user-images.githubusercontent.com/77268333/180717573-eb1e2d68-7038-4e6d-a9b7-e3682e88296d.png)

**Comment by cromewar** [comment, marked as answer]
Hello  @Akhil-franko 
This means is not taking correctly the EtherScan Api key, so here is some recommendations:
1. remember to add `dotenv: .env` into `brownie-config.yaml`
2. check out if you don't have any typo on the API key, use the exact same name as Patrick uses on the course.
3. try to export the Key directly into the terminal before running the code with export keyword, something like this `export KeyName=y378dfskfh`.

I hope this info might help.

**Reply by akhilparakka** [reply]
Got It. Thank you.

**Reply by Python-David** [reply]
If all else fails, I had this problem for days and finally here's what worked for me:

I changed my global compiler version to the version of solidity I declared for my contract: 0.6.6 (from top left...code > preferences > solidity > settings > search solidity > click on solidity configurations > compile using remote version OR just right click on the "pragma solidity..." and select "Change global compiler version")

For some reason, this worked. I'm guessing it has something to do with this from the brownie documentation - WARNING: Make sure all your source files use the same compiler version, otherwise the verification will fail.

**Reply by cromewar** [reply]
Also you can avoid all of this just changing the extension for the hardhat one
